# output for command: export PYTHONPATH=$PWD:$PYTHONPATH && echo $PYTHONPATH && sed -i -e 's/trvaep\.utils/\.\.utils/g' trvaep/model/helper_module.py &&  /apps/gent/RHEL8/haswell-ib/software/Python/3.10.4-GCCcore-11.3.0/bin/python -m pip install --prefix=/scratch/gent/vo/001/gvo00117/easybuild/RHEL8/haswell-ib/software/trvaep/0.1.0-foss-2022a  --no-deps  --ignore-installed  --no-index  --no-build-isolation  .

/tmp/vsc43810/easybuild/build/trvaep/0.1.0/foss-2022a/trvaep/trvaep-0.1.0:/scratch/gent/vo/001/gvo00117/easybuild/RHEL8/haswell-ib/software/trvaep/0.1.0-foss-2022a/lib/python3.10/site-packages:/scratch/gent/vo/001/gvo00117/easybuild/RHEL8/haswell-ib/software/trvaep/0.1.0-foss-2022a/lib/python3.10/site-packages:/apps/gent/RHEL8/haswell-ib/software/torchvision/0.13.1-foss-2022a/lib/python3.10/site-packages:/apps/gent/RHEL8/haswell-ib/software/Pillow-SIMD/9.2.0-GCCcore-11.3.0/lib/python3.10/site-packages:/scratch/gent/vo/001/gvo00117/easybuild/RHEL8/haswell-ib/software/adjustText/0.7.3-foss-2022a/lib/python3.10/site-packages:/scratch/gent/vo/001/gvo00117/easybuild/RHEL8/haswell-ib/software/scanpy/1.9.1-foss-2022a/lib/python3.10/site-packages:/scratch/gent/vo/001/gvo00117/easybuild/RHEL8/haswell-ib/software/umap-learn/0.5.3-foss-2022a/lib/python3.10/site-packages:/apps/gent/RHEL8/haswell-ib/software/leidenalg/0.9.1-foss-2022a/lib/python3.10/site-packages:/apps/gent/RHEL8/haswell-ib/software/python-igraph/0.10.3-foss-2022a/lib/python3.10/site-packages:/apps/gent/RHEL8/haswell-ib/software/PyCairo/1.21.0-GCCcore-11.3.0/lib/python3.10/site-packages:/apps/gent/RHEL8/haswell-ib/software/tqdm/4.64.0-GCCcore-11.3.0/lib/python3.10/site-packages:/apps/gent/RHEL8/haswell-ib/software/scikit-learn/1.1.2-foss-2022a/lib/python3.10/site-packages:/scratch/gent/vo/001/gvo00117/easybuild/RHEL8/haswell-ib/software/statsmodels/0.13.1-foss-2022a/lib/python3.10/site-packages:/scratch/gent/vo/001/gvo00117/easybuild/RHEL8/haswell-ib/software/PyTables/3.8.0-foss-2022a/lib/python3.10/site-packages:/scratch/gent/vo/001/gvo00117/easybuild/RHEL8/haswell-ib/software/scikit-build/0.15.0-GCCcore-11.3.0/lib/python3.10/site-packages:/scratch/gent/vo/001/gvo00117/easybuild/RHEL8/haswell-ib/software/numba/0.56.4-foss-2022a/lib/python3.10/site-packages:/apps/gent/RHEL8/haswell-ib/software/networkx/2.8.4-foss-2022a/lib/python3.10/site-packages:/scratch/gent/vo/001/gvo00117/easybuild/RHEL8/haswell-ib/software/h5py/3.7.0-foss-2022a/lib/python3.10/site-packages:/scratch/gent/vo/001/gvo00117/easybuild/RHEL8/haswell-ib/software/Seaborn/0.12.1-foss-2022a/lib/python3.10/site-packages:/apps/gent/RHEL8/haswell-ib/software/matplotlib/3.5.2-foss-2022a/lib/python3.10/site-packages:/apps/gent/RHEL8/haswell-ib/software/Tkinter/3.10.4-GCCcore-11.3.0/lib/python3.10/site-packages:/apps/gent/RHEL8/haswell-ib/software/PyTorch/1.12.0-foss-2022a/lib/python3.10/site-packages:/apps/gent/RHEL8/haswell-ib/software/expecttest/0.1.3-GCCcore-11.3.0/lib/python3.10/site-packages:/apps/gent/RHEL8/haswell-ib/software/Pillow/9.1.1-GCCcore-11.3.0/lib/python3.10/site-packages:/scratch/gent/vo/001/gvo00117/easybuild/RHEL8/haswell-ib/software/PyYAML/6.0-GCCcore-11.3.0/lib/python3.10/site-packages:/apps/gent/RHEL8/haswell-ib/software/SciPy-bundle/2022.05-foss-2022a/lib/python3.10/site-packages:/apps/gent/RHEL8/haswell-ib/software/pybind11/2.9.2-GCCcore-11.3.0/lib/python3.10/site-packages:/apps/gent/RHEL8/haswell-ib/software/protobuf-python/3.19.4-GCCcore-11.3.0/lib/python3.10/site-packages:/apps/gent/RHEL8/haswell-ib/software/Python/3.10.4-GCCcore-11.3.0/easybuild/python
Processing /tmp/vsc43810/easybuild/build/trvaep/0.1.0/foss-2022a/trvaep/trvaep-0.1.0
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'error'
  error: subprocess-exited-with-error
  
   Preparing metadata (pyproject.toml) did not run successfully.
   exit code: 1
  > [34 lines of output]
      Traceback (most recent call last):
        File "/apps/gent/RHEL8/haswell-ib/software/Python/3.10.4-GCCcore-11.3.0/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
          main()
        File "/apps/gent/RHEL8/haswell-ib/software/Python/3.10.4-GCCcore-11.3.0/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/apps/gent/RHEL8/haswell-ib/software/Python/3.10.4-GCCcore-11.3.0/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 164, in prepare_metadata_for_build_wheel
          return hook(metadata_directory, config_settings)
        File "/apps/gent/RHEL8/haswell-ib/software/Python/3.10.4-GCCcore-11.3.0/lib/python3.10/site-packages/flit_core/buildapi.py", line 49, in prepare_metadata_for_build_wheel
          metadata = make_metadata(module, ini_info)
        File "/apps/gent/RHEL8/haswell-ib/software/Python/3.10.4-GCCcore-11.3.0/lib/python3.10/site-packages/flit_core/common.py", line 408, in make_metadata
          md_dict.update(get_info_from_module(module, ini_info.dynamic_metadata))
        File "/apps/gent/RHEL8/haswell-ib/software/Python/3.10.4-GCCcore-11.3.0/lib/python3.10/site-packages/flit_core/common.py", line 205, in get_info_from_module
          docstring, version = get_docstring_and_version_via_import(target)
        File "/apps/gent/RHEL8/haswell-ib/software/Python/3.10.4-GCCcore-11.3.0/lib/python3.10/site-packages/flit_core/common.py", line 178, in get_docstring_and_version_via_import
          spec.loader.exec_module(m)
        File "<frozen importlib._bootstrap_external>", line 883, in exec_module
        File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
        File "/tmp/vsc43810/easybuild/build/trvaep/0.1.0/foss-2022a/trvaep/trvaep-0.1.0/trvaep/__init__.py", line 3, in <module>
          from .model import *
        File "/tmp/vsc43810/easybuild/build/trvaep/0.1.0/foss-2022a/trvaep/trvaep-0.1.0/trvaep/model/__init__.py", line 2, in <module>
          from .train import Trainer
        File "/tmp/vsc43810/easybuild/build/trvaep/0.1.0/foss-2022a/trvaep/trvaep-0.1.0/trvaep/model/train.py", line 7, in <module>
          from trvaep.model._losses import MSE_kl, mmd
        File "/tmp/vsc43810/easybuild/build/trvaep/0.1.0/foss-2022a/trvaep/trvaep-0.1.0/trvaep/__init__.py", line 18, in <module>
          from get_version import get_version
        File "/scratch/gent/vo/001/gvo00117/easybuild/RHEL8/haswell-ib/software/trvaep/0.1.0-foss-2022a/lib/python3.10/site-packages/get_version/__init__.py", line 290, in <module>
          __version__ = get_version(__file__, vcs="git")
        File "/scratch/gent/vo/001/gvo00117/easybuild/RHEL8/haswell-ib/software/trvaep/0.1.0-foss-2022a/lib/python3.10/site-packages/get_version/__init__.py", line 273, in get_version
          version = method(parent)
        File "/scratch/gent/vo/001/gvo00117/easybuild/RHEL8/haswell-ib/software/trvaep/0.1.0-foss-2022a/lib/python3.10/site-packages/get_version/__init__.py", line 118, in get_version_from_vcs
          vcs_root = find_vcs_root(parent, vcs=vcs)
        File "/scratch/gent/vo/001/gvo00117/easybuild/RHEL8/haswell-ib/software/trvaep/0.1.0-foss-2022a/lib/python3.10/site-packages/get_version/__init__.py", line 146, in find_vcs_root
          from dunamai import _detect_vcs, Vcs
      ModuleNotFoundError: No module named 'dunamai'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

 Encountered error while generating package metadata.
> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
